VirtusLab-Open-Source/strapi-plugin-navigation
# fix: typings issues

## Ticket

https://github.com/VirtusLab-Open-Source/strapi-plugin-navigation/issues/386

## Summary

What does this PR do/solve? 

Typings issues when building plugin's source code injected into Strapi project.

Context: In recent times Strapi `@strapi/helper-plugin` and other where migrated from JS to TS. This move results in typing declaration files being available for Typescript compiler.
Until now `@strapi/helper-plugin` and other eco-system files where not set as dev dependencies and _ts-ignored_ when impored.
In parent project strapi plugins and helpers are installed and TS declaration are available. Source code typings were not matching what Strapi expects so build was failing.

This PR updates some dependencies also.

## Test Plan

- run `yarn`
- run `npm run build`
- build should happen
- copy source code to newly created strapi project as a _static_ plugin
- plugin should be buildable there also
- run Strapi server
- plugin should work as intended, functionalities stay all the same
